Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb

```diff
@@ -142,126 +151,100 @@ def stop
+          @logger.debug("start index: #{start_index} blob size: #{blob_size}")
 
-          # content will be used to calculate the new offset. Create a new variable for processed content.
-          content_length = content.length unless content.nil?
+          content_length = 0
+          blob_reader = BlobReader.new(@logger, @azure_blob, @container, blob_name, file_chunk_size_bytes, start_index, blob_size - 1 - @file_tail_bytes)
 
           is_json_codec = (defined?(LogStash::Codecs::JSON) == 'constant') && (@codec.is_a? LogStash::Codecs::JSON)
-          if (is_json_codec)
-            skip = content.index '{'
-            content.slice!(skip-1) unless (skip.nil? || skip == 0)
-          end #if
+          if is_json_codec
+            parser = JsonParser.new(@logger, blob_reader)
+
+            parser.parse(->(json_content) {
+              content_length = content_length + json_content.length
+
+              enqueue_content(queue, json_content, header, tail)
 
-          if is_json_codec && (@break_json_down_policy != 'do_not_break')
-            @logger.debug("codec is json and policy is not do_not_break")
-            
-            @break_json_batch_count = 1 if @break_json_batch_count <= 0
-            tail = content[-@file_tail_bytes..-1]
-            while (!content.nil? && content.length > @file_tail_bytes)
-              json_event = get_jsons!(content, @break_json_batch_count)
-              break if json_event.nil?
-              @logger.debug("Got json: ========================")
-              @logger.debug("#{json_event[0..50]}...#{json_event[-50..-1]}")
-              @logger.debug("End got json: ========================")
-              @logger.debug("Processed content: #{content[0..50]}...")
-              if @break_json_down_policy == 'with_head_tail'
-                @logger.debug("Adding json head/tails.")
-                json_event = "#{header}#{json_event}#{tail}"
-              end #if
-              @codec.decode(json_event) do |event|
-                decorate(event)
-                queue << event
-              end # decode
-            end
+              on_entry_processed(start_index, content_length, blob_name, new_etag, gen)
+            }, ->(malformed_json) {
+              @logger.debug("Skipping #{malformed_json.length} malformed bytes")
+              content_length = content_length + malformed_json.length
+
+              on_entry_processed(start_index, content_length, blob_name, new_etag, gen)
+            })
           else
-            @logger.debug("Non-json codec or the policy is do not break")
-            # Putting header and content and tail together before pushing into event queue
-            content = "#{header}#{content}" unless header.nil? || header.length == 0
-            @codec.decode(content) do |event|
-              decorate(event)
-              queue << event
-            end # decode
+            begin
```

Comment:
If the codec is not json, you still need to respect that codec. It looks like your change in this else clause is not using the codec at all.